src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -238,13 +212,18 @@ public FireStoreDirectoryEntry retrieveByPath(
 
         String lookupPath = makeLookupPath(fullPath);
 
-        return runTransactionWithRetry(firestore, (xn) -> lookupByFilePath(firestore, collectionId, lookupPath, xn),
+        return fireStoreUtils.runTransactionWithRetry(firestore,
+            xn -> {
+                DocumentSnapshot docSnap = lookupByFilePath(firestore, collectionId, lookupPath, xn);
+                if (docSnap == null) {
+                    return null;
+                }
+                return docSnap.toObject(FireStoreDirectoryEntry.class);
```

Comment:
Does it make sense to have this code inside the retry block? That is, is there a case where it can fail and need to be retried? If not, I think it would read better to compute `DocumentSnapshot` inside the retry, and have code after the retry that converts it to a `FireStoreDirectoryEntry`